### ISSUE TITLE
[Orca] Support multiple outputs of orca estimator predict

### ISSUE THREAD

**Issue by songhappy** [intel-analytics/BigDL#7348]
Current Orca estimator predict only handles model with one output. Many models have multiple outputs, e.g. image object detection models: detection.fasterrcnn_resnet50_fpn, fasterrcnn_mobilenet_v3_large_320_fpn, detection.retinanet_resnet50_fpn the output is a dictionary of "boxes", "labels", "scores", Orca estimator need to handle it correctly.

https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html

**Comment by hkvision** [comment]
What do you think is the reasonable way for this? For example,
- For Spark DataFrame, we output the df with more than one extra columns, how to get the names of the outputs? (Currently the colname is always "prediction") [For SparkXShards of pandas df, same]
- For SparkXShards of dict, the output shard dict should be {"y": [y1, y2, ]}?

**Comment by sgwhat** [comment]
> 



> What do you think is the reasonable way for this? For example,
> 
> * For Spark DataFrame, we output the df with more than one extra columns, how to get the names of the outputs? (Currently the colname is always "prediction") [For SparkXShards of pandas df, same]
> * For SparkXShards of dict, the output shard dict should be {"y": [y1, y2, ]}?

For spark df, maybe we could retrieve the columns' names from the model or users provided?

**Comment by hkvision** [comment]
probably we can add an optimal argument for output_col_names?
Can we always get the names from models? (probably yes?

**Comment by sgwhat** [comment]
> probably we can add an optimal argument for output_col_names? Can we always get the names from models? (probably yes?

I think for tensorflow multi-output model, most cases will have a parameter `output` when creating the model. So it's possible for us to get the names, am I right?

**Comment by hkvision** [comment]
Probably we can use the optimizations in https://github.com/intel-analytics/BigDL/pull/5952 to support multi-outputs as well as enhance performance converting rdd to dataframe with arrow.

**Comment by sgwhat** [comment]
Currently we have supported multi-output prediction model with both pytorch and tensorflow, for the limitations and next step preview plan, we are going to:
- [ ] Support model outputs with different shapes of results.
- [ ] Support automatically allocating output column names.

**Comment by hkvision** [comment]
Seems automatically allocating output column names is not feasible in PyTorch?

**Comment by sgwhat** [comment]
> Seems automatically allocating output column names is not feasible in PyTorch?

For pytorch, if we cannot reach output cols from the model, maybe we could allocate names like `output_1` and `output_2`?

**Comment by hkvision** [comment]
I think we can. We'd better have the same behavior with tf and pytorch. So do think it is reasonable to always output multiple columns in a dataframe whenever users specify output_cols or not?

**Comment by sgwhat** [comment]
> 



> I think we can. We'd better have the same behavior with tf and pytorch. So do think it is reasonable to always output multiple columns in a dataframe whenever users specify output_cols or not?

I think that should depend on the model defined by users, if it's a multi-output model, for example:
```
# tensorflow
model = tf.keras.Model(inputs=[image_input_1, image_input_2], outputs=[score_output, class_output])

# pytorch
class myModel:
    ...
    return y1, y2
```
Whether users specify the output_cols or not, we should return the result dataframe always with multi columns.

**Comment by hkvision** [comment]
So I mean for multiple output model, there will always be multiple columns right?
Seems PyTorch now would stack everything in predictions if output_cols is unspecified.
